### DISCUSSION TITLE
Error in IRHVAC COOLIX protocol? Fan only mode sends dry mode command instead

### DISCUSSION THREAD

**Discussion by mdk2412** [arendst/Tasmota#21541] [Q&A]
I use a wemos D1 mini with tasmota-ir to control my AC using the coolix protocol. Everything works beautifully, with one exception: trying to set fan only mode. To diagnose this issue I used another tasmota-ir device to record what actually was sent. This is what I found:

transmitter: IRHVAC {"Vendor":"COOLIX","Mode":"Fan","Power":"On","FanSpeed":"Auto"}

results in 

receiver: {"IrReceived":{"Protocol":"COOLIX","Bits":24,"Data":"0xB21FC4","DataLSB":"0x4DF823","Repeat":0,"IRHVAC":{"Vendor":"COOLIX","Model":-1,"Command":"Control","Mode":"Dry","Power":"On","Celsius":"On","Temp":25,"FanSpeed":"Auto","SwingV":"Off","SwingH":"Off","Quiet":"Off","Turbo":"Off","Econo":"Off","Light":"Off","Filter":"Off","Clean":"Off","Beep":"Off","Sleep":-1,"iFeel":"Off","SensorTemp":null}}}

transmitter: irsend {"Protocol":"COOLIX","Bits":24,"Data":"0xB2BFE4","DataLSB":"0x4DFD27","Repeat":0}

results in

receiver: {"IrReceived":{"Protocol":"COOLIX","Bits":24,"Data":"0xB2BFE4","DataLSB":"0x4DFD27","Repeat":0,"IRHVAC":{"Vendor":"COOLIX","Model":-1,"Command":"Control","Mode":"Fan","Power":"On","Celsius":"On","Temp":30,"FanSpeed":"Auto","SwingV":"Off","SwingH":"Off","Quiet":"Off","Turbo":"Off","Econo":"Off","Light":"Off","Filter":"Off","Clean":"Off","Beep":"Off","Sleep":-1,"iFeel":"Off","SensorTemp":null}}}

transmitter: IRHVAC {"Vendor":"COOLIX","Mode":"Dry","Power":"On","FanSpeed":"Auto"}

results in

receiver: {"IrReceived":{"Protocol":"COOLIX","Bits":24,"Data":"0xB21FC4","DataLSB":"0x4DF823","Repeat":0,"IRHVAC":{"Vendor":"COOLIX","Model":-1,"Command":"Control","Mode":"Dry","Power":"On","Celsius":"On","Temp":25,"FanSpeed":"Auto","SwingV":"Off","SwingH":"Off","Quiet":"Off","Turbo":"Off","Econo":"Off","Light":"Off","Filter":"Off","Clean":"Off","Beep":"Off","Sleep":-1,"iFeel":"Off","SensorTemp":null}}}

That is, IRHVAC command with "Mode":"Fan" gives the wrong result identical to "Mode:Dry". So I suspect that there is an error in the translation from IRHVAC to actual irsend command somewhere. But where do I look for it?

**Comment by mdk2412** [comment]
Has been solved: https://github.com/crankyoldgit/IRremoteESP8266/pull/2104/commits/c2b5235f8942af847536b70ffe688cbc3bd46a98

**Reply by s-hadinger** [reply]
Thanks. We will wait for a new release of IRRemoteESP8266

**Reply by mdk2412** [reply]
I can confirm that with a self-compiled version of tasmota 14.1.0 including this patch the correct command is sent using IRHVAC.

**Reply by mdk2412** [reply]
Since there has been no new release of IRRemoteESP8266 for so long, could this patch (https://github.com/crankyoldgit/IRremoteESP8266/commit/9785cb910d2b8c81785afb58d76090cb57c5b2df) be included in Tasmota without waiting for the new release?

**Reply by Jason2866** [reply]
@mdk2412 If you provide a PR with this changes, we will merge.
